cpp/arcticdb/storage/mongo/mongo_client.cpp

```diff
@@ -310,31 +360,28 @@ storage::KeySegmentPair MongoClientImpl::read_segment(const std::string &databas
     auto database = client->database(database_name); //TODO maybe cache
     auto collection = database[collection_name];
 
-    try {
-        ARCTICDB_SUBSAMPLE(MongoStorageReadFindOne, 0)
-        auto stream_id = variant_key_id(key);
-        if(StorageFailureSimulator::instance()->configured())
-            StorageFailureSimulator::instance()->go(FailureType::READ);
+    ARCTICDB_SUBSAMPLE(MongoStorageReadFindOne, 0)
+    auto stream_id = variant_key_id(key);
+    if(StorageFailureSimulator::instance()->configured())
+        StorageFailureSimulator::instance()->go(FailureType::READ);
 
+    try {
         auto result = collection.find_one(document{} << "key" << fmt::format("{}", key) << "stream_id" <<
-                                                                                                       fmt::format("{}", stream_id) << finalize);
+                                                     fmt::format("{}", stream_id) << finalize);
         if (result) {
-            const auto &doc = result->view();
+            const auto& doc = result->view();
             auto size = doc["total_size"].get_int64().value;
-            entity::VariantKey stored_key{ detail::variant_key_from_document(doc, key) };
+            entity::VariantKey stored_key{detail::variant_key_from_document(doc, key)};
             util::check(stored_key == key, "Key mismatch: {} != {}");
             return storage::KeySegmentPair(
                     std::move(stored_key),
-                    Segment::from_bytes(const_cast<uint8_t *>(result->view()["data"].get_binary().bytes), std::size_t(size), true)
+                    Segment::from_bytes(const_cast<uint8_t*>(doc["data"].get_binary().bytes), std::size_t(size), true)
             );
         } else {
-            // find_one returned nothing, if this was an exception it would fall through to the catch below.
-            throw std::runtime_error(fmt::format("Missing key in mongo: {} for symbol: {}", key, stream_id));
+            throw KeyNotFoundException(key);
         }
-    }
-    catch(const std::exception& ex) {
-        log::storage().info("Segment read error: {}", ex.what());
-        throw storage::KeyNotFoundException{Composite<VariantKey>{VariantKey{key}}};
+    } catch(const mongocxx::operation_exception& e) {
```

Comment:
We're not catching `std::exception` anymore. What was the thing that used to throw anything inheriting from `std::exception`?